tests/integration/utils.py

```diff
@@ -168,13 +209,51 @@ def __check_attributes_schema(actual_attributes, expected_attributes):
             resource_data = content['data']
             if isinstance(resource_data, list):
                 for resource in resource_data:
-                    __check_resource_schema(resource)
+                    __check_resource_schema(resource, null_value_allowed)
             else:
-                __check_resource_schema(resource_data)
+                __check_resource_schema(resource_data, null_value_allowed)
         elif status_code >= 400:
             errors_data = content['errors']
             self.assertIsInstance(errors_data, list)
             for error in errors_data:
-                __check_error_schema(error)
+                __check_error_schema(error, null_value_allowed)
     except KeyError as error:
         self.fail(error)
+
+
+# Check url for correct base and endpoint, parameters
+def check_url(self, actual_url, endpoint, query_params=None):
+    actual_url_obj = urllib.parse.urlparse(actual_url)
+    base_url_obj = urllib.parse.urlparse(self.base_url)
+    self.assertEqual(actual_url_obj.scheme, base_url_obj.scheme)
+    self.assertEqual(actual_url_obj.netloc, base_url_obj.netloc)
+    self.assertEqual(actual_url_obj.path,
+                     f'{base_url_obj.path}{endpoint}')
+    if query_params:
+        self.assertEqual(dict(urllib.parse.parse_qsl(actual_url_obj.query)),
+                         query_params)
```

Comment:
As we discussed, this block can be improved a bit by combining the same logic into a loop.